swap/src/protocol/bob/swap.rs

```diff
@@ -404,16 +464,24 @@ async fn run_until_internal(
     }
 }
 
-pub async fn request_price_and_setup(
+pub async fn request_price(
     btc: bitcoin::Amount,
     event_loop_handle: &mut EventLoopHandle,
-    execution_params: ExecutionParams,
-    bitcoin_refund_address: bitcoin::Address,
-) -> Result<bob::state::State2> {
+) -> Result<monero::Amount> {
     let xmr = event_loop_handle.request_spot_price(btc).await?;
 
     tracing::info!("Spot price for {} is {}", btc, xmr);
 
+    Ok(xmr)
+}
+
+pub async fn execution_setup(
```

Comment:
Happy to merge clean-up like this into the main branch if you split it out into a dedicated commit!